File: spring_java/src/main/java/dev/pollito/spring_java/sakila/film/domain/port/in/FindAllPortInImpl.java
package dev.pollito.spring_java.sakila.film.domain.port.in;

import dev.pollito.spring_java.sakila.film.domain.model.Film;
import dev.pollito.spring_java.sakila.film.domain.port.out.FindAllPortOut;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class FindAllPortInImpl implements FindAllPortIn {
  private final FindAllPortOut findAllPortOut;

  @Override
  public Page<Film> findAll(Pageable pageable) {
    return findAllPortOut.findAll(pageable);
  }
}

File: spring_kotlin/src/main/kotlin/dev/pollito/spring_kotlin/sakila/film/domain/port/in/FindAllPortInImpl.kt
package dev.pollito.spring_kotlin.sakila.film.domain.port.`in`

import dev.pollito.spring_kotlin.sakila.film.domain.model.Film
import dev.pollito.spring_kotlin.sakila.film.domain.port.out.FindAllPortOut
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Service

@Service
class FindAllPortInImpl(private val findAllPortOut: FindAllPortOut) : FindAllPortIn {
  override fun findAll(pageable: Pageable): Page<Film> = findAllPortOut.findAll(pageable)
}
